staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -138,6 +138,25 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		proxyError(w, req, "service unavailable", http.StatusServiceUnavailable)
 		return
 	}
+
+	// If the ServiceResolver also implements ServiceHostnameResolver, check to see if an external hostname should be used for TLS config.
+	if hostnameResolver, ok := r.serviceResolver.(ServiceHostnameResolver); ok {
+		hostname, isExternal, err := hostnameResolver.ResolveHostname(handlingInfo.serviceNamespace, handlingInfo.serviceName)
+		if err != nil {
+			klog.Errorf("error resolving hostname for %s/%s: %v", handlingInfo.serviceNamespace, handlingInfo.serviceName, err)
```

Comment:
nit: Usually we rather use `runtime.HandleError(err)` which throttles the output. Your function has no real chance to throw and error though (it's informer based).

Reply:
@sttts i was about to change this, and i see where this was coming from now. i copied the existing pattern from line 137 in this file. should i change it there as well?

Reply:
i updated both in https://github.com/kubernetes/kubernetes/pull/123808/commits/310f43410833b5d920cdbf43a883e17dc3c98536